src/GameRoulette/Program.cs

```diff
@@ -28,8 +28,18 @@
 //So the task is to implement new endpoint for buying those knifes
 //Basically we pass an id of a knife we want to buy, and method should returns true if it success or false if it's not
 //Example of a request POST 'localhost:5000/knife/1/1234'
-app.MapPost("/knife/{id}/buy/{price}", (string id, int price) => {
-    //Implement logic here
+app.MapPost("/knife/{id}/buy/{price}", (int id, int price) => {
+    var knife = knifes.FirstOrDefault(x => x.Id == id);
+    if (knife == null)
+    {
+        return false;
+    }
+    if (knife.Price <= price)
+    {
+        knifes.Remove(knife);
+        return true;
+    } 
+    return false;

```

Comment:
You can convert this condition and use ["return early"](https://medium.com/swlh/return-early-pattern-3d18a41bba8) pattern